### PR TITLE
Add elasticSearch-http storage module.

### PR DESCRIPTION
**Pull request by cyberdak**
- Why submit this pull request?
- [x] Bug fix
- [✓] New feature provided
- [x] Improve performance

- Related issues
#773 

___
### New feature or improvement

test ok with elasticsearch-5.6.3

### CODE REVIEW

**Comment by wu-sheng** [comment]
First, welcome your contributions. I and @peng-yongsheng will build an automatic test mechanism to make sure this pr is good enough to merge. So please be patient, this pr will take a little time.

I hope we can work this out together, and merge this finally.

**Comment by peng-yongsheng** [comment]
Wow! What a vast workload. Thx for your contribution. Waiting fix the CI failure.

**Comment by wu-sheng** [comment]
@cyberdak And you should notice, some of your commits are not recognized by GitHub(no avatar commit), and those commits will not shows in project contributions after this pr merged. I think you would like to fix that.

**Comment by cyberdak** [comment]
looks like after i forked this repo it had some change. I need update my repo.

before 5.0.0 GA release, will has anthoer changes happend?

**Comment by wu-sheng** [comment]
@cyberdak 5.0.0 GA is hard to say. It is a long way from here. The most important thing of merging this pr is only related to collector automatic tests.

Most things between 5.0.0 alpha -> GA are about optimization and end user experience.

**Comment by cyberdak** [comment]
CI check ready.
which files need license modify?

**Comment by wu-sheng** [comment]
All `.java`, `pom.xml` and other config files, which I think you already have.

**Comment by wu-sheng** [comment]
I think the CI and license are easy things. After we release the 5.0.0 alpha, we will work with you for how to test this pr. Is that OK for you?

**Comment by cyberdak** [comment]
it is ok.

**Comment by wu-sheng** [comment]
And conflict need to be fixed.

**Comment by peng-yongsheng** [comment]
+1

That's a good idea, modifications are so much. I'm worrying about how to check those codes and how to demonstrate those are no problem.
ShardingJDBC team will pull request their implement in the near future. It could use with the same way.

**Comment by wu-sheng** [comment]
@chidaodezhongsheng If you are still doing this, please consider our suggestion. 

But from sharding jdbc, you should be care of mysql driver license, which is GPL v2. This can not be part of distribution. 

FYI @terrymanu

**Comment by cyberdak** [comment]
1. it is ok.
2.document is ok.In fact ,the http module is copy from es module then change impl to http.Every class is some in es module.So the support is easy.
3.I think it is impossible in few days . i think should treat me as a free contribuor not a worker in 58. The new feature is non of 58's business. If i am not a employee in a company , so what?  I think the core question is the module are so large and need some test.I will take care the tests.

**Comment by wu-sheng** [comment]
> 3.I think it is impossible in few days . i think should treat me as a free contribuor not a worker in 58. The new feature is non of 58's business. If i am not a employee in a company , so what? I think the core question is the module are so large and need some test.I will take care the tests.

We always treat every contributor as individual. The major concern is about maintaining. I am not worried about providing a feature with bug( every one's codes have bugs ), instead, I am worried have no one to fix bugs in time.

Another reason I asked for public news, is about the end user cases of this features. The community is driven by the end users, right?

Hope I explain myself clear. :)

So if you want to try, we can accept this as an incubating feature first. But then, you have the following responsibilities:
1. Take part in test and Apache release process, because after merged, this feature is part of it.
1. Sync the changes in our default es implementations.

The incubating feature at least last 3-6 month, if this feature is stable, has public user case for this feature, and you are still active, we can change this into core feature, even default implementation. 

@cyberdak Is this good enough for you?

If so, you should work on resolve conflicts and documents, including settings, usage, and log to tell people how to change the implementation to ES-http.

And please test es-http storage after you sync the master's codes. 

BTW, I am definitely sure, there are several new type(table)s are going to add in beta.

File: apm-collector/apm-collector-boot/src/main/resources/application.yml [hunk 1]
```diff
@@ -34,18 +34,27 @@ ui:
...
-#    user_name: sa
+#  elasticsearch:
+#    cluster_name: elasticsearch
+#    cluster_transport_sniffer: true
+#    cluster_nodes: 10.126.94.37:9300
+#    index_shards_number: 2
+#    index_replicas_number: 0
+#    ttl: 7
+#storage:
+#  elasticsearch-http:
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-collector/apm-collector-boot/src/main/resources/application.yml]: Setting about this should not stay in documents. Please add this as a part of `Advanced Features` at here: https://github.com/apache/incubator-skywalking/blob/master/docs/README.md  

English document is required, Chinese is optional for now.

[the remark this answers is not in the document]
Comment (wu-sheng) [reply apm-collector/apm-collector-boot/src/main/resources/application.yml]: Make sure your title includes `incubating`, but doc file name didn't need that.

**Comment by cyberdak** [comment]
I will resolve the CI and update code sync with es module.
Document is provided now.

**Comment by wu-sheng** [comment]
@peng-yongsheng This pr includes in a new dependency, https://github.com/searchbox-io/Jest, please be attention.

**Comment by wu-sheng** [comment]
@cyberdak Related pr #1063 proposed, please see. If we agree with all things we have discussed, please consider to 
1. Add `(Incubating)` in your document title
1. Solve conflict
1. Follow my previous comments, e.g. remove the setting in `application.yml`, even in comments.
1. Passed the CI

Then I will start to merge. @cyberdak You can test in your company env, and provide us more feedback about this feature.

**Comment by wu-sheng** [comment]
@cyberdak Any update?

**Comment by wu-sheng** [comment]
Closing this, no update more than 3 months.

**Pull request closed without merging**